### ISSUE TITLE
Replace `ward` with a maintained alternative

### ISSUE THREAD

**Issue by doronbehar** [petereon/beaupy#126]
Hello,

First of all, I must say I am very glad I found this library, it fits great for my usecase, so thanks for sharing it. Now to the main topic:

`ward` was archived on 1st of October. Not only that, in my distribution we marked `ward` as broken due to this issue:

- https://github.com/darrenburns/ward/issues/380

The same comment can be applied to all other dependencies of this project maintained by you - `python-yakh` and `questo`.

**Comment by petereon** [comment]
Heya,

thanks for the kind words, really glad you are enjoying this library.

For the matter at hand - I am painfully aware of this and also have been trying to avoid it for as long as possible.  
  
`beaupy` is quite heavily tested and rework for another framework will be time-consuming, but what's worse, due to some funky input stream interplay, `yakh` doesn't play well with `pytest` (the only obvious alternative).

But I appreciate it's about time I dealt with this, so, considering the issue you reference is probably blocking, I will handle this in a two-step process:
- I'll extract the dev and test dependencies into a separate file (`test-requirements.txt` or so), so that they are not included in the dependencies when it's being installed for all of the libraries you mention. This can be expected to be done soon - I'll try to allocate time in the upcoming weekend but no promises
- I'll migrate the tests to `pytest` - on a longer time horizon

**Comment by doronbehar** [comment]
Thanks man. Please update me when the dependencies are updated, so I'll be able to update the Nixpkgs PR for your packages:

- https://github.com/NixOS/nixpkgs/pull/366468

**Comment by petereon** [comment]
Will do!

**Comment by petereon** [comment]
@doronbehar - soo, I made some changes, as of `beaupy==3.10` tests still depend on `ward` but it's no longer a part of `pyproject.toml` - not sure if that helps. I'll try to migrate the tests towards `pytest` in the upcoming weeks.

**Comment by doronbehar** [comment]
Hmm well, perhaps I wasn't clear enough when I opened the issue. The fact the currently latest release uses ward only stops our distribution from running the tests, not from building and using the package. If I understand it correctly, only `[tool.poetry.dependencies]` matter for our purposes.

**Comment by petereon** [comment]
Right, sorry, misunderstood. I've reworked tests across packages to `pytest` (thanks LLMs). Do you need me to release each or can you work with it as is?

**Comment by doronbehar** [comment]
I tested all the packages with the pytest tests and it all seems good :). I hope these are not false positives. A new release will be nice :). Thanks!

**Comment by petereon** [comment]
Got around to release [3.10.1](https://pypi.org/project/beaupy/3.10.1/)

**Comment by doronbehar** [comment]
Thanks for the release - also the new releases for `questo` and `python-yakh` was also comfortable.
